cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -80,10 +81,15 @@
 
   public static final String APP_CONFIG_HEADER = "X-App-Config";
 
-  protected int getInstances(HttpRequest request) throws IllegalArgumentException, JsonSyntaxException {
-    Instances instances = parseBody(request, Instances.class);
+  protected int getInstances(HttpRequest request) throws BadRequestException {
+    Instances instances;
+      try {
+        instances = parseBody(request, Instances.class);
+      } catch (JsonSyntaxException e) {
+        throw new BadRequestException("Invalid JSON in request");
```

Comment:
would be good to know why JSON was invalid. `JsonSyntaxException` might have it, right?